Remove shared subscriptions prefix when matching topics

atlantis opened the pull request:
My MQTT broker (VerneMQ) allows shared subscriptions even when running v3... though I believe technically they're a v5 feature (https://www.hivemq.com/blog/mqtt5-essentials-part7-shared-subscriptions/).

I'm honestly not thrilled about this particular implementation - it was the easy route but requires a regex on each receive.  A better solution would probably be to re-architect subscriptions and separate the topic from the filter (since those can be different in the case of shared subscriptions).

So mostly just throwing this out there to see what you think.

atlantis committed:
Remove shared subscriptions prefix when matching topics

My MQTT broker (VerneMQ) allows shared subscriptions even when running v3... though I believe technically they're a v5 feature.

stakach commented:
probably better to do something like

```crystal

filter_array = filter.split('/')
filter_array = filter_array[2..-1] if filter_array.first.starts_with?("$")

```

Add a spec too and I'll merge :)

atlantis committed:
Refactor and spec

atlantis committed:
Call static function explicitly

Huh seems odd that an object-level method can't find the class-level methods automatically, but oh well...

stakach committed:
fix(v3 client): shared subscription detection

as all shared subscriptions start with `$share`
